import "DataCommon.proto";
import "ProtoHero.proto";

package ProtoAchievement;

message AchievementCPP
{
	required uint32 id			= 1;  //任务id
	required uint32 status		= 2;  //任务状态
	repeated uint32 current		= 3;   //任务当前进度
}

message AchievementAllCPP
{
	repeated AchievementCPP  achievement	= 1;
}

//前端触发成就验证
message VerifyAchievementReq
{
	required uint32 type	= 1; //成就类型
}

//推送成就的请求
message PushAchievementReq
{
	required AchievementAllCPP achievements		= 1;
}

//领取成就奖励
message GetAchievementRewardReq
{
	required uint32 id		= 1;
}

message GetAchievementRewardResp
{
	optional DataCommon.EquipmentAllCPP equips		= 1; //装备
	optional DataCommon.CommonBaseRewardCPP	base	= 2; //资源、钻石和各种次数
	optional ProtoHero.HeroBaseInfoCPP	hero		= 3; //解锁的英雄信息
	required AchievementCPP achievement				= 4; //成就信息	
}
